repose-aggregator/src/docs/asciidoc/recipes/functional-test-framework.adoc

```diff
@@ -0,0 +1,163 @@
+The `repose-test.properties` file tells that framework what version of Repose, directories, host, and ports to use.
+It should be in the https://en.wikipedia.org/wiki/.properties#Format[Java properties file format] like the following example:
+
+----
+test.root.directory=/default/test/root # <1>
+
+repose.version=${version} # <2>
+repose.container=valve # <3>
+repose.home=/default/repose/home # <4>
+repose.config.directory=/configs # <5>
+repose.config.templates=/default/config/templates # <6>
+repose.log.name=/logs/repose.log # <7>
+repose.lint.log.name=/logs/repose-lint.log # <8>
+repose.log.pattern=/logs/repose_%d{yyyy-MM-dd_HHmmss}.log # <9>
+
+repose.jar=/default/valve/location # <10>
+repose.lint.jar=/default/lint/location # <11>
+
+repose.root.war=/default/root/war/location # <12>
+glassfish.jar=/default/glassfish/location # <13>
+tomcat.jar=/default/tomcat/location # <14>
+mocks.war=/default/mocks/war/location # <15>
+
+target.hostname=localhost # <16>
+
+port.finder.port.start=10000 # <17>
+port.finder.port.max=65535 # <18>
+port.finder.port.range=100 # <19>
+port.finder.property.name.worker.id=org.gradle.test.worker # <20>
+----
+<1> The absolute path of the root directory to run tests under.
+The user running the tests must have permission to read and write in this directory.
+<2> The version of Repose.
+<3> The servlet container to run Repose in.
+A value of `valve` will run a standalone instance of Repose using an embedded container.
+<4> The absolute path of the home directory which will contain Repose artifacts, and possibly other Repose test structures.
+<5> The path of the directory relative to the `test.root.directory` or `repose.home` directory which will contain Repose configuration files.
+<6> The absolute path of the directory which contains the configuration file templates.
+<7> The Repose log file path relative to the `test.root.directory` or `repose.home` directory.
+<8> The Repose Lint log file path relative to the `test.root.directory` or `repose.home`.
+<9> The log filename path pattern relative to the `test.root.directory` or `repose.home` directory.
+<10> The absolute path to the Repose JAR file.
+<11> The absolute path of the Repose Lint JAR file.
+<12> The absolute path of the Repose `ROOT.WAR` file.
+<13> The absolute path of the Glassfish JAR file.
+<14> The absolute path of the Tomcat JAR file.
+<15> The absolute path of the WAR file containing mock services.
+<16> The target hostname (usually `localhost`).
+<17> The port number to start at when scanning for open ports.
+<18> The max port number which can be used when scanning for open ports.
+<19> The step size between workers when scanning for open ports.
+<20> The name of a System property the value of which will be used as the worker ID.
+
+[NOTE]
+====
+Not all properties are strictly required.
+In fact, most properties have reasonable defaults.
+However, any property which takes an absolute path should be given a value.
+====
```

Comment:
I think we need to be more specific about this and possibly put this note upfront, not after the fact.